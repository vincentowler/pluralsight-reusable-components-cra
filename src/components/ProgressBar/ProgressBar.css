.container {
  border: solid 1px lightgray;
  width: 100px;
}
