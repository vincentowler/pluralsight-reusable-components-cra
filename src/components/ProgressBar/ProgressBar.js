import React, { PropTypes } from 'react';
import styles from './ProgressBar.css';

/**
 * Display progress from 0% to 100%
 */
class ProgressBar extends React.Component {
  constructor(props) {
    super(props);
    this.getBarColor = this.getBarColor.bind(this);
  }

  getBarColor() {
    if (this.props.percent == 100) {
      return 'green';
    } else if (this.props.percent > 50) {
      return 'lightgreen';
    }
    return 'red';
  }

  render() {
    const height = this.props.height || 5;

    let barStyle = {
      width: this.props.percent,
      height: height,
      backgroundColor: this.getBarColor()
    };

    return (
      <div className={styles.container}>
        <div style={barStyle} />
      </div>
    );
  }
}

ProgressBar.propTypes = {
  /**
   * Number from 0 and 100 that represents progress.
   */
  percent: PropTypes.number.isRequired,
  /**
   * Bar height in pixels.
   */
  height: PropTypes.number
};

export default ProgressBar;
