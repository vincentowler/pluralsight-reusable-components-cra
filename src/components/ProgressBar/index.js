export ProgressBar from './ProgressBar.js';
