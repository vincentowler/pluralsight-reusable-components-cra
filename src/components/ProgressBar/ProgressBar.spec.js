import React from 'react';
import renderer from 'react-test-renderer';
import ProgressBar from './ProgressBar';

// Snapshot test
describe('ProgressBar', () => {
  test('should render red with 20%', () => {
    const component = renderer.create(<ProgressBar percent={20} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should render light green with 60%', () => {
    const component = renderer.create(<ProgressBar percent={60} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should render dark green with 100%', () => {
    const component = renderer.create(<ProgressBar percent={100} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
