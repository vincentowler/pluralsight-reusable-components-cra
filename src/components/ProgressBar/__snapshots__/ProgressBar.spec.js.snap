exports[`ProgressBar should render dark green with 100% 1`] = `
<div
  className={undefined}>
  <div
    style={
      Object {
        "backgroundColor": "green",
        "height": 5,
        "width": 100,
      }
    } />
</div>
`;

exports[`ProgressBar should render light green with 60% 1`] = `
<div
  className={undefined}>
  <div
    style={
      Object {
        "backgroundColor": "lightgreen",
        "height": 5,
        "width": 60,
      }
    } />
</div>
`;

exports[`ProgressBar should render red with 20% 1`] = `
<div
  className={undefined}>
  <div
    style={
      Object {
        "backgroundColor": "red",
        "height": 5,
        "width": 20,
      }
    } />
</div>
`;
